### PR TITLE
=htp a few small things that got lost in the #17988 merge

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-docs-dev/rst/java/code/docs/http/javadsl/PathDirectiveExampleTest.java [hunk 1]
```diff
@@ -82,9 +79,4 @@ public RouteResult apply(RequestContext ctx, Integer userId) {
         );
         //#path-examples
     }
-
-    // FIXME: remove once #17988 is merged
-    public static <T> Route handleWith1(RequestVal<T> val, Object o) {
-        return null;
-    }
```
Comment (ktoso) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/PathDirectiveExampleTest.java]: ah, great! :-)

**Comment by ktoso** [comment]
LGTM, thanks Johannes

**Comment by jrudolph** [comment]
Unrelated test failure.

**Comment by jrudolph** [comment]
PLS REBUILD

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
:)

**Comment by jrudolph** [comment]
ah :)

**Comment by jrudolph** [comment]
almost

**Comment by ktoso** [comment]
LGTM, thanks - will merge in a sec.

**Comment by jrudolph** [comment]
Rebased.

**Comment by jrudolph** [comment]
Error lines in https://jenkins.akka.io/job/pr-validator-per-commit-jenkins/3286/:

```
  8412 Test akka.stream.javadsl.FlowTest.mustBeAbleToRecover failed: java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg while waiting for 1, took 3.025 sec
  8413     at scala.Predef$.assert(Predef.scala:179)
  8414     at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:338)
  8415     at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:315)
  8416     at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
  8417     at akka.testkit.JavaTestKit.expectMsgEquals(JavaTestKit.java:378)
  8418     at akka.stream.javadsl.FlowTest.mustBeAbleToRecover(FlowTest.java:548)
  8419     ...
  8679 Test akka.stream.javadsl.SourceTest.mustBeAbleToRecover failed: java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg while waiting for 1, took 3.014 sec
  8680     at scala.Predef$.assert(Predef.scala:179)
  8681     at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:338)
  8682     at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:315)
  8683     at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
  8684     at akka.testkit.JavaTestKit.expectMsgEquals(JavaTestKit.java:378)
  8685     at akka.stream.javadsl.SourceTest.mustBeAbleToRecover(SourceTest.java:549)
  8686     ...
 10619 Failed: Total 867, Failed 2, Errors 0, Passed 865, Pending 22
 10620 Failed tests:
 10621 	akka.stream.javadsl.SourceTest
 10622 	akka.stream.javadsl.FlowTest
 13983 (akka-stream-tests-experimental/test:test) sbt.TestsFailedException: Tests unsuccessful
 13984 Total time: 832 s, completed Jul 24, 2015 2:24:33 PM
```

**Comment by jrudolph** [comment]
(unrelated failures)

**Comment by ktoso** [comment]
LGTM thanks, yeap failures unrelated and already fixed.

**Pull request merged**